sdk/scala-sdk/src/main/scala/com/akkaserverless/scalasdk/EntityOptions.scala

```diff
@@ -38,11 +40,11 @@ trait EntityOptions extends ComponentOptions {
    * @return
    *   the headers requested to be forwarded as metadata (cannot be mutated, use withForwardHeaders)
    */
-  def forwardHeaders(): java.util.Set[String]
+  def forwardHeaders: immutable.Set[String]
 
   /**
    * Ask Akka Serverless to forward these headers from the incoming request as metadata headers for the incoming
    * commands. By default no headers except "X-Server-Timing" are forwarded.
    */
-  def withForwardHeaders(headers: java.util.Set[String]): ComponentOptions
+  def withForwardHeaders(headers: immutable.Set[String]): EntityOptions
```

Comment:
this is the main reason for adding a scala sdk ComponentOptions. Scala developers will pass a Scala Set